src/dcp/dcp.c

```diff
@@ -491,17 +497,20 @@ int main(int argc, char** argv)
     /* connect to DAOS source pool if uuid is valid */
     if (mfu_src_file->type == DAOS) {
         /* if DAOS source pool uuid is valid, then set source file type to DAOS */
-        daos_connect(rank, svc, src_pool_uuid, src_cont_uuid, &src_poh, &src_coh); 
+        daos_connect(rank, src_svc, src_pool_uuid, src_cont_uuid, &src_poh, &src_coh); 
     }
 
     if (mfu_dst_file->type == DAOS) {
         if (daos_uuid_valid(dst_pool_uuid) && !same_pool) {
             /* if DAOS is the source and destination type, and containers are in different pools,
              * then connect to the second pool */
-            daos_connect(rank, svc, dst_pool_uuid, dst_cont_uuid, &dst_poh, &dst_coh); 
+            daos_connect(rank, dst_svc, dst_pool_uuid, dst_cont_uuid, &dst_poh, &dst_coh); 
         } else {
-            /* if DAOS is source and destination type, and containers are in the same pool,
-             * then pool is already connected, so we just need to open and/or create the container */
+            /* Containers are using the same pool uuid.
+             * This is unlikely to ever happen but we can print an error just in case */
+            if (strcmp(dst_svc, src_svc) == 0) {
```

Comment:
Do you want ``!=0`` here?

And if one of these pointers happens to be NULL, is it valid to call strcmp?

Reply:
I thought we should also check the compound if condition ``if (daos_uuid_valid(dst_pool_uuid) && !same_pool)``.  There are two ways to get to the else, either dst_pool_uuid is not valid or same_pool is set.

Are we always doing the right thing in this else block for the case that dst_pool_uuid is not valid?

Reply:
> I thought we should also check the compound if condition `if (daos_uuid_valid(dst_pool_uuid) && !same_pool)`. There are two ways to get to the else, either dst_pool_uuid is not valid or same_pool is set.
> 
> Are we always doing the right thing in this else block for the case that dst_pool_uuid is not valid?

@adammoody I removed the check for a valid dest uuid since if the type is set the pool uuid is already parsed/checked, and it would fail before this if it wasn't valid. And yes, I did mean for the check to be != 0 👍

Reply:
@adammoody ok, which part specifically? I saw that I originally had some tabs in there, and removed them. But I might have missed one?

Reply:
Actually, it looks good now.  Wonder if my browser just messed up the formatting somehow.  Anyway, looks good to me.  Good to you?

Reply:
@adammoody yep